File: app/src/main/java/com/example/liftoff/extractor/email/OutlookExtractor.java
package com.example.liftoff.extractor.email;

import com.example.liftoff.storage.filesystem.FileStorage;
import com.microsoft.graph.serviceclient.GraphServiceClient;

import java.util.Properties;

public class OutlookExtractor extends EmailExtractor{
    private GraphServiceClient client = null;

    public OutlookExtractor(String emailAddress, String userToken, FileStorage storage) {
        super(emailAddress, userToken, storage);
    }

    private void initializeGraph(final Properties oAuthProperties) {
        // TODO implement
        //this.client = new GraphServiceClient();
    }

    @Override
    public void authenticate() throws Exception {
        final Properties oAuthProperties = new Properties();
        initializeGraph(oAuthProperties);
    }

    @Override
    public void extract() {
        // TODO implement
        // GET /me/mailFolders/{id}/messages/{id}/attachments/{id}
        //this.client.me().mailFolders()
    }

    @Override
    public void close() throws Exception {
        // TODO impelment cleanup
    }
}

File: src/main/scala/com/example/liftoff/extractor/email/outlook/OutlookExtractor.scala
package com.example.liftoff.extractor.email.outlook

import com.example.liftoff.error.ExtractorError
import com.example.liftoff.extractor.email.EmailExtractor
import com.example.liftoff.storage.Storage
import zio.IO

class OutlookExtractor(emailAddress: String, userToken: String, storage: Storage) extends EmailExtractor(emailAddress: String, userToken: String, storage: Storage) {

  override def authenticate(): IO[ExtractorError, Unit] = ???

  override def close(): IO[ExtractorError, Unit] = ???

  override def extract(): IO[ExtractorError, Unit] = ???
}
